homeassistant/components/stream/__init__.py

```diff
@@ -112,22 +92,20 @@ def shutdown(event):
 class Stream:
     """Represents a single stream."""
 
-    def __init__(self, hass, source, options=None):
+    def __init__(
+        self, hass: HomeAssistant, source: str, options: dict[str, str | bool]
+    ) -> None:
         """Initialize a stream."""
-        self.hass = hass
-        self.source = source
-        self.options = options
-        self.keepalive = False
-        self.access_token = None
-        self._thread = None
+        self.hass: HomeAssistant = hass
+        self.source: str = source
+        self.keepalive: bool = False
+        self.access_token: str | None = None
+        self._thread: threading.Thread | None = None
         self._thread_quit = threading.Event()
-        self._outputs: dict[str, StreamOutput] = {}
-        self._fast_restart_once = False
-
-        if self.options is None:
-            self.options = {}
-
-        self.ll_hls = self.options.pop(CONF_LL_HLS, False)
+        self._outputs: dict[str, HlsStreamOutput | RecorderOutput] = {}
+        self._fast_restart_once: bool = False
+        self.ll_hls: bool = cast(bool, options.pop(CONF_LL_HLS, False))
+        self.options: dict[str, str] = cast(Dict[str, str], options)
```

Comment:
I think `cast` can be also removed here, but this needs testing.
```suggestion
        self.ll_hls: bool = options.pop(CONF_LL_HLS, False)
        self.options: dict[str, str] = options
```